### PR TITLE
Angle grinder AGAIN! Also, makes plasma cutters used for salvage instead of mining (and base gearpacks)

### PR DESCRIPTION
**Pull request by Thera-Pissed**
## About The Pull Request
re-implements #1716 as a base and adds cutter functionality to plasma cutters
also implements and uses gear packs from #2103 that never got in.
makes plasma cutters unable to mine, shorter range, and do burn instead of brute.
angle grinders are integrated into grinder packs, like water backtanks. Also draws power from a cell.
jackhammers can no longer break down walls and girders.
### Cutters can currently deconstruct:
- girders
- walls
- reinforced walls
- airlocks
- airlock frames
- grilles
- machine frames
- computer frames
- catwalks
- chairs
- beds
- tables
- lockers & crates
- salvage machines (the wasteplanet ones)
- railings
- lattice
- wooden barricades

cutters can also cut open safes and locked lockers & crates

attempted to revert map changes but using git checkout doesn't completely revert the gecko. Will try again later.

adds a mech salvage saw and a prototype energy saw. Doesn't make either available yet.

sprites by me
image

https://github.com/shiptest-ss13/Shiptest/assets/90987989/12262338-055f-4c7c-86d1-d31279ab953c

## Why It's Good For The Game
Jackhammers as a main salvage tool is dumb, and angle grinders and cutters make more sense and have better functionality. Cutting apart old ruins and ships should be a reasonable source of material and shouldn't take years.

## Changelog

:cl:
add: angle grinders for salvage
add: reworks plasma cutters for salvage
/:cl:

### CODE REVIEW

**Comment by Martinpachu** [comment]
very rad

**Comment by Apogee-dev** [comment]
Can you remove breaching functionality from jackhammers, since there's now a specialty tool for that job? I never really liked the thing with jackhammers battering down metal walls anyway.

**Comment by Martinpachu** [comment]
> Can you remove breaching functionality from jackhammers, since there's now a specialty tool for that job? I never really liked the

should replace them on the colossus and any other ships in that case

**Comment by Martinpachu** [comment]
any other ships that use them for breaching i mean. i dont think any are using them for mining though.

**Comment by Thera-Pissed** [comment]
I'll remove the jackhammer effect, though angle packs are cheap enough that I don't want to touch maps just yet. Which ships  would need their jackhammers replaced, though?

**Comment by Martinpachu** [comment]
> I'll remove the jackhammer effect, though angle packs are cheap enough that I don't want to touch maps just yet. Which ships would need their jackhammers replaced, though?

colossus, hound (if you even care.....) gecko, hyena, inkwell and chronicle.

**Comment by Thera-Pissed** [comment]
couple maps tweaked to have roundstart cutters and packs

**Comment by JimKil3** [comment]
the mudskipper has a surgical saw in a "salvage supplies" crate iirc which i'm pretty confident is meant to represent an angle grinder

File: code/modules/clothing/ears/_ears.dm [hunk 1]
```diff
@@ -15,6 +15,7 @@
 	name = "earmuffs"
 	desc = "Protects your hearing from loud noises, and quiet ones as well."
 	icon_state = "earmuffs"
+	slot_flags = ITEM_SLOT_EARS
```
Comment (FalloutFalcon) [inline code/modules/clothing/ears/_ears.dm]: pointless

File: code/game/objects/structures/window.dm [hunk 2]
```diff
@@ -286,6 +287,15 @@
 	qdel(src)
 	update_nearby_icons()
 
+/obj/structure/window/deconstruct_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(!I.tool_start_check(user, amount=0))
+		return FALSE
+	if (I.use_tool(src, user, decon_time, volume=100))
+		to_chat(user, "<span class='warning'>You shatter [src] with the [I].</span>")
```
Comment (FalloutFalcon) [inline code/game/objects/structures/window.dm]:
```suggestion
		to_chat(user, span_warning("You shatter [src] with the [I]."))
```

File: code/game/objects/structures/tables_racks.dm [hunk 3]
```diff
@@ -227,6 +227,15 @@
 	else
 		return ..()
 
+/obj/structure/table/deconstruct_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(!I.tool_start_check(user, amount=0))
+		return FALSE
+	if (I.use_tool(src, user, 1 SECONDS, volume=0))
+		to_chat(user, "<span class='warning'>You cut [src] into sheets.</span>")
```
Comment (FalloutFalcon) [inline code/game/objects/structures/tables_racks.dm]:
```suggestion
		to_chat(user, span_warning("You cut [src] into sheets."))
```

File: code/game/objects/structures/girders.dm [hunk 4]
```diff
@@ -162,7 +143,7 @@
 						S.use(1)
 						to_chat(user, "<span class='notice'>You fully reinforce the wall.</span>")
 						var/turf/T = get_turf(src)
-						T.PlaceOnTop(/turf/closed/wall/r_wall)
+						T.PlaceOnTop(/turf/closed/)
```
Comment (FalloutFalcon) [inline code/game/objects/structures/girders.dm]:
```suggestion
						T.PlaceOnTop(/turf/closed)
```

File: code/modules/jobs/access.dm [hunk 5]
```diff
@@ -117,6 +117,8 @@
 	if (gen_ship_access(ship))
 		return TRUE
 
+	if(!item)
+		return FALSE
```
Comment (FalloutFalcon) [inline code/modules/jobs/access.dm]: Already opened a seperate pr for this.

File: code/modules/mining/equipment/angle_grinder.dm [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	AddComponent(/datum/component/two_handed)
+	AddElement(/datum/element/tool_flash, 2)
+	AddElement(/datum/element/tool_bang, 2)
+
+/// triggered on wield of two handed item
+/obj/item/attachment/anglegrinder/proc/on_wield(obj/item/source, mob/user)
+	SIGNAL_HANDLER
+
+	playsound(src, 'sound/weapons/chainsawhit.ogg', 100, TRUE)
+	force = 24
```
Comment (FalloutFalcon) [inline code/modules/mining/equipment/angle_grinder.dm]: mabye add a force_unwielded and force_wielded var for these

File: code/modules/projectiles/guns/energy/special.dm [hunk 7]
```diff
@@ -127,10 +127,10 @@
 
 	heat = 3800
 	usesound = list('sound/items/welder.ogg', 'sound/items/welder2.ogg')
-	tool_behaviour = TOOL_WELDER
-	toolspeed = 0.7 //plasmacutters can be used as welders, and are faster than standard welders
+	tool_behaviour = TOOL_DECONSTRUCT
```
Comment (FalloutFalcon) [inline code/modules/projectiles/guns/energy/special.dm:125]: should prob communicate with gristlebee about #3145..

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]
Good stuff, really support the plasma cutter changes
Where your checking for two separate tools is really asking for us to port tools having multiple tool qualities. Remind me at some point to work on that augh.
Could use some span defines on some stuff but I don't care all that much,

File: code/modules/projectiles/guns/energy/special.dm (hunk 7 above)
Comment (Thera-Pissed) [reply code/modules/projectiles/guns/energy/special.dm:125]: communicated, going to wait until the PR is in to do discussed tweaks

**Comment by Thera-Pissed** [comment]
waiting for #3145 to be merged before I continue working on this, since it overlaps a decent amount.

**Comment by Thera-Pissed** [comment]
couple oddball things I want to work on but Im happy with the basic grinder and plasma cutter changes. not sure about plasma cutters in toolbelts from the destructible walls pr, but I'll wait and see how it goes.

File: code/game/objects/structures/girders.dm [hunk 8]
```diff
@@ -162,7 +143,7 @@
 						S.use(1)
 						to_chat(user, "<span class='notice'>You fully reinforce the wall.</span>")
 						var/turf/T = get_turf(src)
-						T.PlaceOnTop(/turf/closed/wall/r_wall)
+						T.PlaceOnTop(/turf/closed)
```
Comment (FalloutFalcon) [inline code/game/objects/structures/girders.dm]:
```suggestion
						T.PlaceOnTop(/turf/closed/wall/r_wall)
```

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]

**Review by FalloutFalcon — APPROVED** [review]

**Pull request merged**